Mods/src/PuppiMod.cc

```diff
@@ -141,235 +160,222 @@ void PuppiMod::SlaveBegin()
     }
   }
   fNumEtaBins = fMaxEtas.size();
-  std::cout << "Number of Eta bins: " << fNumEtaBins << std::endl;
+  Info("SlaveBegin", "Number of Eta bins: %d", fNumEtaBins);
+
+  fMapper = new ParticleMapper(fR0, fR0, fMaxEtas[fNumEtaBins-1]);
 }
 
 //--------------------------------------------------------------------------------------------------
-void PuppiMod::SlaveTerminate()
+void
+PuppiMod::SlaveTerminate()
 {
   // ===== deallocate memory ====
-  // Or you know, don't
+  delete fPuppiParticles;
+  delete fMapper;
 }
 
 //--------------------------------------------------------------------------------------------------
-void PuppiMod::Process()
+void
+PuppiMod::Process()
 {
-  // Process entries of the tree. 
-  fVertexes = GetObject<VertexCol>(fVertexesName);
-  fPFCandidates = GetObject<PFCandidateCol>(fPFCandidatesName);
+  // Process entries of the tree.
+  auto* vertexes = GetObject<VertexCol>(fVertexesName);
+  if (!vertexes)
+    SendError(kAbortAnalysis, "Process", "PV collection not found");
+
+  auto* pv = vertexes->At(0);
 
-  PFCandidateArr *PuppiParticles = new PFCandidateArr;        // This array is only used if adding object to event, not publishing
+  auto* pfCandidates = GetObject<PFCandidateCol>(fPFCandidatesName);
 
   if (fDumpingPuppi) {
-    std::cout << "Primary Vertex location: " << fVertexes->At(0)->Position().x() << ", ";
-    std::cout << fVertexes->At(0)->Position().y() << ", " << fVertexes->At(0)->Position().z() << std::endl;
+    std::cout << "Primary Vertex location: " << pv->Position().x() << ", ";
+    std::cout << pv->Position().y() << ", " << pv->Position().z() << std::endl;
   }
 
   // This mapper will return particles that are only close in eta, phi space
-  ParticleMapper *Mapper = new ParticleMapper();
-  Mapper->Initialize(*fPFCandidates,fR0,fR0,fMaxEtas[fNumEtaBins-1]);
```

Comment:
Two things: 1. As discussed in the class declaration already, it is better to reuse the mapper rather than create one in each event. 2. You never delete this mapper! Always follow a new by a delete. The original code must have been allocating space for N_{eta-phi bins} + N_{particles} integers for each event without ever releasing it.